### PR TITLE
feat: add update indicator

### PR DESCRIPTION
**Pull request by manuel-rw**
<br/>
<div align="center">
  
  <h3>Homarr</h3>
</div>

**Thank you for your contribution. Please ensure that your pull request meets the following pull request:**

- [x] Builds without warnings or errors (``pnpm buid``, autofix with ``pnpm format:fix``)
- [x] Pull request targets ``dev`` branch
- [x] Commits follow the [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] No shorthand variable names are used (eg. ``x``, ``y``, ``i`` or any abbrevation)

image

### CODE REVIEW

File: apps/nextjs/src/components/user-avatar-menu.tsx [hunk 1]
```diff
@@ -64,6 +68,21 @@ export const UserAvatarMenu = ({ children }: UserAvatarMenuProps) => {
     // We use keepMounted so we can add event listeners to prevent navigating away without saving the board
     <Menu width={300} withArrow withinPortal keepMounted>
       <Menu.Dropdown>
+        {availableUpdates && availableUpdates.length > 0 && availableUpdates[0] && (
```
Comment (Meierschlumpf) [inline apps/nextjs/src/components/user-avatar-menu.tsx:71]: Maybe show the available updates only the admin? Or what do you think?

File: packages/api/src/router/update-checker.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { createSubPubChannel } from "../../../redis/src/lib/channel";
```
Comment (Meierschlumpf) [inline packages/api/src/router/update-checker.ts:1]: Is it possible to import `@homarr/redis` here instead?

File: packages/cron-jobs/src/jobs/update-checker.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import { compareSemVer, isValidSemVer } from "semver-parser";
+
+import { EVERY_HOUR } from "@homarr/cron-jobs-core/expressions";
+import { logger } from "@homarr/log";
+
+import packageJson from "../../../../package.json";
+import { createSubPubChannel } from "../../../redis/src/lib/channel";
+import { createCronJob } from "../lib";
+
+export const updateCheckerJob = createCronJob("updateChecker", EVERY_HOUR, {
```
Comment (Meierschlumpf) [inline packages/cron-jobs/src/jobs/update-checker.ts:11]: What about using a `request handler` instead?

File: packages/translation/src/lang/en.json [hunk 4]
```diff
@@ -733,7 +733,8 @@
         "logout": "Logout",
         "login": "Login",
         "homeBoard": "Your home board",
-        "loggedOut": "Logged out"
+        "loggedOut": "Logged out",
+        "updateAvailable": "({countUpdates}) updates available: {tag}"
```
Comment (Meierschlumpf) [inline packages/translation/src/lang/en.json:737]: IMO move the brackets `5 updates available (v1.2.3)` makes more sense

**Review by Meierschlumpf — CHANGES_REQUESTED** [review]

File: packages/cron-jobs/src/jobs/update-checker.ts (hunk 3 above)
Comment (manuel-rw) [reply packages/cron-jobs/src/jobs/update-checker.ts:11]: All other cron jobs for non-integrations do not use it. Why should we here? Isn't it exclusively for the integrations?

Comment (Meierschlumpf) [reply packages/cron-jobs/src/jobs/update-checker.ts:11]: Your correct, right now we only use it with the cached-integration-request-handler, but I build it that way that all requests can be handled centralized and cached. For example my plan is to move the ping results to this system as well. You can simply use the createCachedRequestHandler method provided

File: apps/nextjs/src/components/user-avatar-menu.tsx [hunk 5]
```diff
@@ -60,10 +64,27 @@ export const UserAvatarMenu = ({ children }: UserAvatarMenuProps) => {
     });
   }, [logoutUrl, openModal, router]);
 
+  const isAdmin = session.data?.user.permissions.includes("admin");
+
   return (
     // We use keepMounted so we can add event listeners to prevent navigating away without saving the board
     <Menu width={300} withArrow withinPortal keepMounted>
       <Menu.Dropdown>
+        {isAdmin && availableUpdates && availableUpdates.length > 0 && availableUpdates[0] && (
```
Comment (Meierschlumpf) [inline apps/nextjs/src/components/user-avatar-menu.tsx:73]: If we only show them to the admin we should also only show the indicator then. So it's probably easier if we only fetch it in the user.tsx if the user is admin and then pass RouterOutputs[*] | undefined down. Then the logic for available updates does not need to change

File: packages/cron-jobs/src/jobs/update-checker.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
+import { Octokit } from "octokit";
+import { compareSemVer, isValidSemVer } from "semver-parser";
+
+import { EVERY_HOUR } from "@homarr/cron-jobs-core/expressions";
+import { logger } from "@homarr/log";
+
+import packageJson from "../../../../package.json";
+import { createSubPubChannel } from "../../../redis/src/lib/channel";
```
Comment (Meierschlumpf) [inline packages/cron-jobs/src/jobs/update-checker.ts:8]: Wrong import for channel

**Review by Meierschlumpf — CHANGES_REQUESTED** [review]

**Review by Meierschlumpf — APPROVED** [review]

**Pull request merged**